joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -149,6 +151,12 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
         } else {
           (Option(baseIdentifier), Option(_typeFullName))
         }
+      case IdentifierName =>
+        val typeFullName = scope.findTypeFromMethodName(nameFromNode(expression)) match {
```

Comment:
What is an example of this? Shouldn't we consider rather popping of the stack to find the first instance of a type that has a method that shares this name? That way we can ensure we aren't in trouble from shadowing.

Reply:
https://github.com/Privado-Inc/joern/blob/46674de7d30a8e6a9f08f30af8ad00177e6fe27e/joern-cli/frontends/csharpsrc2cpg/src/test/scala/io/joern/csharpsrc2cpg/querying/ast/MethodTests.scala#L80

Refer this test.

Reply:
You're right, but this would fail in case of multiple inheritance where we'll push multiple types on the stack and we might not get the required type when we pop the stack.

Reply:
I'm re-structuring this as part of [#4117](https://github.com/joernio/joern/issues/4117) so I wouldn't block you on this, the tests will fail if I break something here.